src/cmn/timeoutManager.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * timeoutManager.cpp
+ *
+ *  Created on: Feb 16, 2020
+ *      Author: Anjana_Sreekumar
+ */
+
+#include "timeoutManager.h"
+#include "log.h"
+
+#include <cstring>
+#include <stdio.h>
+#include <sys/timerfd.h>
+#include <unistd.h>
+
+TimeoutManager::Timer::Timer() : fd_m(0)
```

Comment:
i am not convinced with benefit we have with Timer class under TimeroutManager. How does it help ?
May be keeping fd and required things under TimeoutManager might look easy.

Reply:
The timer class encapsulates the posix timer functions. Hence kept it as separate class. And since it is tightly coupled with timeout manager kept the class under timeout manager. Let me know if I need to keep it out?